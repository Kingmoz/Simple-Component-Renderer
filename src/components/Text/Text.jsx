import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Wrapper = styled.p`
  ${props => props.css ? props.css : null}
`;

const Text = ({ content, css }) => {
  return (
    <Wrapper css={css}>
      {content ? content : 'Content is missing'}
    </Wrapper>
  );
};

Text.proptypes = {
  content: PropTypes.string.isRequired,
  css: PropTypes.object
}

export default Text;