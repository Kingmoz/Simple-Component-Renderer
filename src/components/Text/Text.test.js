import React from 'react';
import { mount } from 'enzyme';

import Text from './Text';

function setup() {
  const props = {
    content: 'message'
  };

  return mount(<Text {...props} />);
}

describe('Text', () => {
  const wrapper = setup();

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render content correctly', () => {
    expect(wrapper.text()).toEqual('message');
  });

});