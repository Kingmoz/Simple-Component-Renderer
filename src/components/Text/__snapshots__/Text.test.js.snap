// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render properly 1`] = `"<p class=\\"sc-bdVaJa ibTDPh\\">message</p>"`;
